src/analyses/base.ml

```diff
@@ -2075,6 +2075,7 @@ struct
       let s2_a = address_from_value s2_v in
       let s2_typ = AD.type_of s2_a in
       (* compute value in string literals domain if s1 and s2 are both string literals *)
+      (* TODO: comparing types structurally should not be done (use typSig instead!) *)
```

Comment:
@nathanschmidt Please compare `typeSig` here instead of types; in this specific case it is fine, but in general types can be cyclic.

Reply:
`CilType.Typ.equal` is fine too. Possibly even better because it doesn't allocate visitor objects on each call.